src/modules/Dropdown/DropdownMessage.js

```diff
@@ -0,0 +1,61 @@
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import {
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  META,
+} from '../../lib'
+
+/**
+ * A message sub-component for Dropdown component.
+ */
+function DropdownMessage(props) {
+  const { className, noResultsMessage, type, value } = props
+
+  const classes = cx('message', className)
+
+  let message = ''
+  if (type === 'noResultsMessage') {
+    message = noResultsMessage
+  } else {
+    message = `Max ${value} selection${value > 1 ? 's' : ''}`
```

Comment:
As I mentioned abowe, let's use predefined props there.

```jsx
DropdownMessage.defaultProps = {
  noResults: 'No results found.',
  maxSelections: 'Max {value} selection{suffix}',
}

DropdownMessage.propTypes = {
  noResults: customPropTypes.contentShorthand,
  maxSelections: customPropTypes.contentShorthand,
}
```

```jsx
import { cloneElement, isValidElement } from 'react'

const createMessage = (props, type, value) => {
  const message = props[type]

  if(isValidElement(message)) return cloneElement(message, { type, value })
  if(typeof message !== 'string') return message
  return message
    .replace('value', value)
    .replace('suffix', suffix)
}
```

Also let's add an example with localization to confirm that my suggestions are not falsy.